dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java

```diff
@@ -60,4 +60,6 @@ public class TaskDTO {
     public Set<String> dependants = new HashSet<>();
 
     public Set<String> dependencies = new HashSet<>();
+
+    public List<String> previousTaskNames = new ArrayList<>();
```

Comment:
I don't think there is a need for this field. There is a field dependencies, which are the direct dependencies.